src/coreclr/jit/lowerxarch.cpp

```diff
@@ -6365,6 +6365,9 @@ void Lowering::ContainCheckIntrinsic(GenTreeOp* node)
             TryMakeSrcContainedOrRegOptional(node, op1);
         }
     }
+
+    // It's expected to be folded to 0/1 earlier
+    assert(intrinsicName != NI_System_Runtime_CompilerServices_RuntimeHelpers_IsKnownConstant);
```

Comment:
The fact that morph no longer folds all of these means something downstream from it must, otherwise disabling constant folding from VN and on becomes a correctness problem.

Reply:
Good point, let me see what I can do